crates/repository/src/lib.rs

```diff
@@ -85,6 +86,51 @@ impl Repository {
         Ok(path_with_maximal_modified_time.to_path_buf())
     }
 
+    pub fn check_new_version_available(
+        &self,
+        own_version: &str,
+        path: impl AsRef<Path>,
+    ) -> Result<Option<(Version, Version)>> {
+        #[derive(Deserialize, Debug)]
+        struct Cargo {
+            package: Package,
+        }
+        #[derive(Deserialize, Debug)]
+        struct Package {
+            version: String,
+        }
```

Comment:
Why is this needed? Does it make sense to make this public i.e. move it to types?

Reply:
These types are needed for serde to know the structure we want to deserialize.
Alternatives include a long chain of getting fields, asserting their type and the associated error handling or finding a crate that can parse the whole Cargo.toml.
I think this is the simplest method.
Unless we intend to parse more stuff from our Crgo tomls in the future, I don't think it makes sense to move these anywhere else.

Reply:
Okay, makes sense. Let's keep it there.